src/iexamine.cpp

```diff
@@ -3748,64 +3748,100 @@ void iexamine::pay_gas( player &p, const tripoint &examp )
     }
 }
 
-void iexamine::climb_down( player &p, const tripoint &examp )
+void iexamine::ledge( player &p, const tripoint &examp )
 {
-    if( !g->m.has_zlevels() ) {
-        // No climbing down in 2D mode
-        return;
-    }
 
-    if( !g->m.valid_move( p.pos(), examp, false, true ) ) {
-        // Covered with something
-        return;
-    }
+    uilist cmenu;
+    cmenu.text = _( "There is a ledge here.  What do you want to do?" );
+    cmenu.addentry( 1, true, 'j', _( "Jump over." ) );
+    cmenu.addentry( 2, true, 'c', _( "Climb down." ) );
+    cmenu.query();
 
-    tripoint where = examp;
-    tripoint below = examp;
-    below.z--;
-    while( g->m.valid_move( where, below, false, true ) ) {
-        where.z--;
-        below.z--;
-    }
+    switch( cmenu.ret ) {
+        case 1: {
+            tripoint dest( p.posx() + 2 * sgn( examp.x - p.posx() ), p.posy() + 2 * sgn( examp.y - p.posy() ),
+                           p.posz() );
+            if( p.get_str() < 4 ) {
+                add_msg( m_warning, _( "You are too weak to jump over an obstacle." ) );
+            } else if( p.get_str() < 4 ) {
+                add_msg( m_warning, _( "You are too weak to jump over an obstacle." ) );
```

Comment:
This looks like a duplicate of the line above

Reply:
Thanks.